cabal-install/Distribution/Client/Status.hs

```diff
@@ -0,0 +1,417 @@
+                                              , package
+                                              )
+import Distribution.InstalledPackageInfo      ( InstalledPackageInfo(..) )
+import Distribution.Package
+import Distribution.Verbosity
+
+import qualified Distribution.Simple.Setup as Cabal
+import qualified Distribution.Simple.LocalBuildInfo as LBI
+
+import qualified Paths_cabal_install          ( version )
+
+import Distribution.Client.Compat.ExecutablePath ( getExecutablePath )
+import System.Directory                       ( doesFileExist )
+import System.FilePath                        ( splitFileName
+                                              , combine
+                                              , splitPath
+                                              , joinPath
+                                              , takeDirectory
+                                              , (<.>)
+                                              )
+import Control.Monad                          ( when
+                                              , forM
+                                              , guard
+                                              , forM_
+                                              )
+import Data.Version                           ( showVersion
+                                              )
+import Data.Maybe                             ( isJust
+                                              )
+import Data.List                              ( groupBy
+                                              , sortBy
+                                              )
+import Data.Ord                               ( comparing
+                                              )
+import Text.PrettyPrint                       ( empty
+                                              , ($$)
+                                              , vcat
+                                              , nest
+                                              , text
+                                              , punctuate
+                                              , ($+$)
+                                              , (<+>)
+                                              )
+import Data.Monoid                            ( (<>)
+                                              )
+import qualified Data.Monoid as Monoid
+import Data.List                              ( inits )
+import Data.Foldable                          ( asum )
+
+
+
+status :: Verbosity -> GlobalFlags -> StatusFlags -> IO ()
+status verbosity globalFlags statusFlags = do
+  (useSandbox, config) <- loadConfigOrSandboxConfig
+    verbosity
+    (globalFlags { globalRequireSandbox = Cabal.Flag False })
+  (defaultComp, platform, defaultConf) <- configCompilerAux'
+                                        $ savedConfigureFlags config
+  let distPref = useDistPref defaultSetupScriptOptions
+  -- TODO: this currently only works if the cabal version used for configuring
```

Comment:
This TODO is actually quite difficult to fix. If the LBI is from a different version of Cabal library, as it might be in the case of a Custom setup script, we can't actually get our grubby hands on it. This problem plagues package-tests. Probably the only way to fix it is to have a different program which links against the relevant version of Cabal.